crates/sui-macros/src/lib.rs

```diff
@@ -171,6 +189,33 @@ pub fn register_fail_point_if(
     register_fail_point_impl(identifier, Arc::new(move || Box::new(callback())));
 }
 
+/// Register code to run locally if the fail point is hit, with a value provided
+/// by the test. If the registered callback returns a Some(v), then the `v` is
+/// passed to the callback in the test.
+///
+/// In the test:
+///
+/// ```ignore
+///     register_fail_point_arg("foo", || {
+///         Some(42)
+///     });
+/// ```
+///
+/// In the code:
+///
+/// ```ignore
+///     let mut value = 0;
+///     fail_point_arg!("foo", |arg| {
+///        value = arg;
+///     });
+/// ```
+pub fn register_fail_point_arg<T: Send + 'static>(
```

Comment:
This is amazing! If the two places have different types, where does it fail? The downcast in get_fp_some_result?

Reply:
yes, the downcast will fail at runtime